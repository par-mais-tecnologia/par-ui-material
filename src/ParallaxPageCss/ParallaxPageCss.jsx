import React from 'react'
import { withStyles } from '@material-ui/core'
const styles = theme => (theme.parallax)

const ParallaxPageCss = (props) => {
  const { classes } = props

  return (
    <div className={
      props.mobile
        ? [classes.parallaxContainer, classes.mobile].join(' ') : classes.parallaxContainer
    }>
      <div className={classes.parallaxSection}>
        <div className={classes.parallaxContent}>
          {props.header}
        </div>
      </div>
      <div className={classes.content}>
        {props.content}
      </div>
      <div>
        {props.footer}
      </div>
    </div>
  )
}

export default withStyles(styles)(ParallaxPageCss)
