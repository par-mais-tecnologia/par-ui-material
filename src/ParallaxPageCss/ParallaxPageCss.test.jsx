import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import ParallaxPageCss from './'

configure({ adapter: new Adapter() })

const foo = (
  <div className='flex flex-column items-center justify-center pa4' style={{ backgroundColor: 'red', height: '100%' }}>
    <div>00000000</div>
    <div>aaaaaaaaa</div>
    <div>uuuuuuuuuuu</div>
  </div>
)

export const testShallow = describe('Stepper shallow', () => {
  const getShallow = () => (
    <ParallaxPageCss>
      {foo}
    </ParallaxPageCss>
  )

  it(`shallow ParallaxPageCss`, () => {
    const wrapper = shallow(getShallow())
    expect(wrapper).toBeTruthy()
  })

  it('shallow ParallaxPageCss match Snapshot', () => {
    const wrapper = shallow(getShallow())
    expect(wrapper).toMatchSnapshot()
  })
})
