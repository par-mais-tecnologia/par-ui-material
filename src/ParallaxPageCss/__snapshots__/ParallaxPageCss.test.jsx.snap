// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper shallow shallow ParallaxPageCss match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(ParallaxPageCss)>
    <div
      className="flex flex-column items-center justify-center pa4"
      style={
        Object {
          "backgroundColor": "red",
          "height": "100%",
        }
      }
    >
      <div>
        00000000
      </div>
      <div>
        aaaaaaaaa
      </div>
      <div>
        uuuuuuuuuuu
      </div>
    </div>
  </WithStyles(ParallaxPageCss)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="flex flex-column items-center justify-center pa4"
        style={
          Object {
            "backgroundColor": "red",
            "height": "100%",
          }
        }
      >
        <div>
          00000000
        </div>
        <div>
          aaaaaaaaa
        </div>
        <div>
          uuuuuuuuuuu
        </div>
      </div>,
      "classes": Object {},
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            00000000
          </div>,
          <div>
            aaaaaaaaa
          </div>,
          <div>
            uuuuuuuuuuu
          </div>,
        ],
        "className": "flex flex-column items-center justify-center pa4",
        "style": Object {
          "backgroundColor": "red",
          "height": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "00000000",
          },
          "ref": null,
          "rendered": "00000000",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "aaaaaaaaa",
          },
          "ref": null,
          "rendered": "aaaaaaaaa",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "uuuuuuuuuuu",
          },
          "ref": null,
          "rendered": "uuuuuuuuuuu",
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="flex flex-column items-center justify-center pa4"
          style={
            Object {
              "backgroundColor": "red",
              "height": "100%",
            }
          }
        >
          <div>
            00000000
          </div>
          <div>
            aaaaaaaaa
          </div>
          <div>
            uuuuuuuuuuu
          </div>
        </div>,
        "classes": Object {},
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              00000000
            </div>,
            <div>
              aaaaaaaaa
            </div>,
            <div>
              uuuuuuuuuuu
            </div>,
          ],
          "className": "flex flex-column items-center justify-center pa4",
          "style": Object {
            "backgroundColor": "red",
            "height": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "00000000",
            },
            "ref": null,
            "rendered": "00000000",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "aaaaaaaaa",
            },
            "ref": null,
            "rendered": "aaaaaaaaa",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "uuuuuuuuuuu",
            },
            "ref": null,
            "rendered": "uuuuuuuuuuu",
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
